core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -37,42 +37,162 @@
+      new Builder(Name.LOGS_DIR)
+          .setDefaultValue(String.format("${%s}/logs", Name.WORK_DIR))
+          .build();
   public static final PropertyKey METRICS_CONF_FILE =
-      create(Name.METRICS_CONF_FILE, String.format("${%s}/metrics.properties", Name.CONF_DIR));
+      new Builder(Name.METRICS_CONF_FILE)
+          .setDefaultValue(String.format("${%s}/metrics.properties", Name.CONF_DIR))
+          .build();
   public static final PropertyKey NETWORK_HOST_RESOLUTION_TIMEOUT_MS =
-      create(Name.NETWORK_HOST_RESOLUTION_TIMEOUT_MS, "5sec");
+      new Builder(Name.NETWORK_HOST_RESOLUTION_TIMEOUT_MS)
+          .setDefaultValue("5sec")
+          .build();
   public static final PropertyKey NETWORK_NETTY_HEARTBEAT_TIMEOUT_MS =
-      create(Name.NETWORK_NETTY_HEARTBEAT_TIMEOUT_MS, "30sec");
+      new Builder(Name.NETWORK_NETTY_HEARTBEAT_TIMEOUT_MS)
+          .setDefaultValue("30sec")
+          .build();
   public static final PropertyKey NETWORK_THRIFT_FRAME_SIZE_BYTES_MAX =
-      create(Name.NETWORK_THRIFT_FRAME_SIZE_BYTES_MAX, "16MB");
+      new Builder(Name.NETWORK_THRIFT_FRAME_SIZE_BYTES_MAX)
+          .setDefaultValue("16MB")
+          .build();
   public static final PropertyKey SITE_CONF_DIR =
-      create(Name.SITE_CONF_DIR, "${user.home}/.alluxio/,/etc/alluxio/");
-
-  public static final PropertyKey TEST_MODE = create(Name.TEST_MODE, false);
-  public static final PropertyKey VERSION = create(Name.VERSION, ProjectConstants.VERSION);
-  public static final PropertyKey WEB_RESOURCES = create(Name.WEB_RESOURCES,
-      String.format("${%s}/core/server/common/src/main/webapp", Name.HOME));
-  public static final PropertyKey WEB_THREADS = create(Name.WEB_THREADS, 1);
+      new Builder(Name.SITE_CONF_DIR)
+          .setDefaultValue("${user.home}/.alluxio/,/etc/alluxio/")
+          .build();
+  public static final PropertyKey TEST_MODE =
+      new Builder(Name.TEST_MODE)
+          .setDefaultValue(false)
+          .build();
+  public static final PropertyKey VERSION =
+      new Builder(Name.VERSION)
+          .setDefaultValue(ProjectConstants.VERSION)
+          .build();
+  public static final PropertyKey WEB_RESOURCES =
+      new Builder(Name.WEB_RESOURCES)
+          .setDefaultValue(String.format("${%s}/core/server/common/src/main/webapp", Name.HOME))
+          .build();
+  public static final PropertyKey WEB_THREADS =
+      new Builder(Name.WEB_THREADS)
+          .setDefaultValue(1)
+          .build();
   public static final PropertyKey WORK_DIR =
-      create(Name.WORK_DIR, String.format("${%s}", Name.HOME));
-  public static final PropertyKey ZOOKEEPER_ADDRESS = create(Name.ZOOKEEPER_ADDRESS, null);
+      new Builder(Name.WORK_DIR)
+          .setDefaultValue(String.format("${%s}", Name.HOME))
+          .build();
+  public static final PropertyKey ZOOKEEPER_ADDRESS =
+      new Builder(Name.ZOOKEEPER_ADDRESS)
+          .setDefaultValue(null)
```

Comment:
Do we need default value here? If so, could you explicitly specify the default value in the other null defaults?